internal/pkg/util/exec/pipe_linux.go

```diff
@@ -62,26 +61,26 @@ func setPipe(data []byte) (*os.File, error) {
 			sylog.Warningf("\"echo 65536 > /proc/sys/net/core/wmem_default\"")
 		}
 	} else {
-		return nil, fmt.Errorf("failed to determine current pipe size: %s", err)
+		return -1, fmt.Errorf("failed to determine current pipe size: %s", err)
 	}
 
 	pipeFd, err := syscall.Dup(fd[1])
 	if err != nil {
-		return nil, fmt.Errorf("failed to duplicate pipe file descriptor: %s", err)
+		return -1, fmt.Errorf("failed to duplicate pipe file descriptor: %s", err)
 	}
 
 	if n, err := syscall.Write(fd[0], data); err != nil || n != len(data) {
-		return nil, fmt.Errorf("failed to write data to stdin: %s", err)
+		return -1, fmt.Errorf("failed to write data to stdin: %s", err)
 	}
 
-	return os.NewFile(uintptr(pipeFd), "pipefd"), err
```

Comment:
So without associated `*os.File` fd is not closed by the runtime?

Reply:
Yes